src/SFML/Window/DRM/InputImplUDev.cpp

```diff
@@ -0,0 +1,719 @@
+                else if (inputEvent.type == EV_ABS)
+                {
+                    switch (inputEvent.code)
+                    {
+                    case ABS_MT_SLOT:
+                        currentSlot = inputEvent.value;
+                        touchFd = *itr;
+                        break;
+                    case ABS_MT_TRACKING_ID:
+                        atSlot(currentSlot).id = inputEvent.value;
+                        touchFd = *itr;
+                        break;
+                    case ABS_MT_POSITION_X:
+                        atSlot(currentSlot).pos.x = inputEvent.value;
+                        touchFd = *itr;
+                        break;
+                    case ABS_MT_POSITION_Y:
+                        atSlot(currentSlot).pos.y = inputEvent.value;
+                        touchFd = *itr;
+                        break;
+                    }
+                }
+                else if (inputEvent.type == EV_SYN && inputEvent.code == SYN_REPORT &&
+                          *itr == touchFd)
+                {
+                    // This pushes events directly to the queue, because it
+                    // can generate more than one event.
+                    processSlots();
+                }
+
+                bytesRead = read(*itr, &inputEvent, sizeof(inputEvent));
+            }
+
+            if ((bytesRead < 0) && (errno != EAGAIN))
+                sf::err() << " Error: " << std::strerror(errno) << std::endl;
+        }
+
+        // Finally check if there is a Text event on stdin
+        //
+        // We only clear the ICANON flag for the time of reading
+
+        newTerminalConfig.c_lflag &= ~(tcflag_t)ICANON;
+        tcsetattr(STDIN_FILENO, TCSANOW, &newTerminalConfig);
+
+        timeval timeout;
+        timeout.tv_sec = 0;
+        timeout.tv_usec = 0;
+
+        unsigned char code = 0;
+
+        fd_set readFDSet;
+        FD_ZERO(&readFDSet);
+        FD_SET(STDIN_FILENO, &readFDSet);
+        int ready = select(STDIN_FILENO + 1, &readFDSet, NULL, NULL, &timeout);
+
+        if (ready > 0 && FD_ISSET(STDIN_FILENO, &readFDSet))
+            bytesRead = read(STDIN_FILENO, &code, 1);
+
+        if ((code == 127) || (code == 8))  // Suppress 127 (DEL) to 8 (BACKSPACE)
+            code = 0;
+        else if (code == 27)  // ESC
+        {
+            // Suppress ANSI escape sequences
+            FD_ZERO(&readFDSet);
+            FD_SET(STDIN_FILENO, &readFDSet);
+            ready = select(STDIN_FILENO + 1, &readFDSet, NULL, NULL, &timeout);
+            if (ready > 0 && FD_ISSET(STDIN_FILENO, &readFDSet))
+            {
+                unsigned char tempBuffer[16];
+                bytesRead = read(STDIN_FILENO, tempBuffer, 16);
+                code = 0;
+            }
+        }
+
+        newTerminalConfig.c_lflag |= ICANON;
+        tcsetattr(STDIN_FILENO, TCSANOW, &newTerminalConfig);
+
+        if (code > 0)
+        {
+            // TODO: Proper unicode handling
```

Comment:
Do we need to create an issue to track this? What does "proper unicode handling" mean or rather, what is "improper" with the current approach?

Reply:
Hi there,

This code is taking an 8-bit char and copying it blindly into the 32-bit "unicode" field of an sfml text event.

This worked for me for what I needed but I expect it won't work for anything other than ascii characters (thus the comment).

Reply:
Thanks @mickelson 
@eXpl0it3r I sketched out a codepoint combiner. Wold that form be ok? This as of now assumes UTF-8 set as keyboard layout. Would you like me to continue it here or open a separate PR once this is done? It's not all, encoding needs to be recognised and ANSI needs to be handled aswell I guess.
```cpp
if (code > 0)
{
    if (code < 0x80) // 1 byte long UTF-8
    {
        event.type = sf::Event::TextEntered;
        event.text.unicode = code;
        return true;
    }
    else if (code < 0xBF)  // following UTF-8 bytes
    {
        uniBytes.push_back(code);
        sf::Uint32 uniChar;
        sf::Utf8::toUtf32(uniBytes.begin(), uniBytes.end(), &uniChar);
        uniSize--;
        if (uniSize == 0)  // last UTF-8 byte
        {
            uniBytes.clear();
            event.type = sf::Event::TextEntered;
            event.text.unicode = uniChar;
            return true;
        }
        return false;
    }
    else if (code < 0xE0) // 2 bytes long UTF-8
    {
        uniSize = 1;
        uniBytes.push_back(code);
        return false;
    }
    else if (code < 0xF0) // 3 bytes long UTF-8
    {
        uniSize = 2;
        uniBytes.push_back(code);
        return false;
    }
    else if (code < 0xF8) // 4 bytes long UTF-8
    {
        uniSize = 3;
        uniBytes.push_back(code);
        return false;
    }
    else
    {
        return false;
    }
}
```

Reply:
As it's quite a bit of code and I have a feeling that there's a simpler/better way to do it, than custom parsing, I'd like to move this into a separate PR.